kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -72,19 +91,48 @@ def check(self, _):
                 self._context_limit,
             )
 
-        self.report_highwater_offsets(highwater_offsets, self._context_limit)
+        self.report_highwater_offsets(highwater_offsets, self._context_limit, cluster_id)
         self.report_consumer_offsets_and_lag(
-            consumer_offsets, highwater_offsets, self._context_limit - len(highwater_offsets)
+            consumer_offsets,
+            highwater_offsets,
+            self._context_limit - len(highwater_offsets),
+            broker_timestamps,
+            cluster_id,
         )
         if self.config._close_admin_client:
             self.client.close_admin_client()
 
-    def report_highwater_offsets(self, highwater_offsets, contexts_limit):
+    def _load_broker_timestamps(self, cluster_id):
+        """Loads broker timestamps from persistent cache."""
+        broker_timestamps = defaultdict(dict)
+        try:
+            for topic_partition, content in json.loads(
+                read_persistent_cache("broker_timestamps_" + cluster_id)
```

Comment:
Using the methods in the base class would simplify things quite a bit since you would not need to handle the keys

Reply:
I switched to the base class methods, but I think it's better to keep the cluster id in the key, to avoid loading wrong data. What do you think?

Reply:
I might be wrong but you should not be worried about this because we concatenate the `check_id` with the key you provided and use this as the "real" key in the cache: https://github.com/DataDog/integrations-core/blob/d0dc80b1cf2c01ba3840db4ac0b8c1101f4d07cc/datadog_checks_base/datadog_checks/base/checks/base.py#L1009-L1011

The check_id is a hash of the instance configuration so we should not load stuff from another cluster